packages/lib-core/src/types/common.ts

```diff
@@ -0,0 +1,12 @@
+import type { AnyObject } from '@monorepo/common';
+
+// Type for extension hook
+export type ExtensionHook<TType, TOptions extends AnyObject = AnyObject> = (
+  options: TOptions,
+) => [TType, boolean, unknown];
+
+export type ExtensionK8sResourceIdentifier = {
```

Comment:
Please add the following comment above:
```
// TODO(vojtech): use types like RequiredProperties<T, K> and OptionalProperties<T, K>
// to change optionality of specific properties, instead of redefining them via "&" intersection
```